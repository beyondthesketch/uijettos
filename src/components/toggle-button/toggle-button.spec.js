import React from 'react';
import renderer, {act} from 'react-test-renderer';

import UijettosToggleButton from './toggle-button';

describe('UijettosToggleButton component', () => {
    test('Renders correctly when not supplied any props', () => {
        const tree = renderer.create(
                <UijettosToggleButton />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when given a cssClassPrefix and state is false', () => {
        const tree = renderer.create(
                <UijettosToggleButton
                    initialState={ false }
                    cssClassPrefix="doodads"
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when given a cssClassPrefix and state is true', () => {
        const tree = renderer.create(
                <UijettosToggleButton
                    initialState={ true }
                    cssClassPrefix="doodads"
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when given initialState of true', () => {
        const tree = renderer
            .create(
                <UijettosToggleButton
                    initialState={ true }
                />
            )
            .toJSON();


        expect(tree).toMatchSnapshot();
    });

    test('Calls whenClicked correctly when given initialState of true', () => {
        const mockWhenClicked = jest.fn();

        const tree = renderer
            .create(
                <UijettosToggleButton
                    initialState={ true }
                    whenClicked={ mockWhenClicked }
                />
            )
            .toJSON();

        // toggle off
        act(
            () => tree.props.onClick()
        );
        expect(mockWhenClicked).toHaveBeenCalledWith(false);
    });

    test('Renders correctly when given initialState of false', () => {
        const tree = renderer
            .create(
                <UijettosToggleButton
                    initialState={ false }
                />
            )
            .toJSON();


        expect(tree).toMatchSnapshot();
    });


    test('Calls whenClicked correctly when given initialState of false', () => {
        const mockWhenClicked = jest.fn();

        const tree = renderer
            .create(
                <UijettosToggleButton
                    initialState={ false }
                    whenClicked={ mockWhenClicked }
                />
            )
            .toJSON();

        // toggle on
        act(
            () => tree.props.onClick()
        );
        expect(mockWhenClicked).toHaveBeenCalledWith(true);
    });

    test('Renders correctly when given a labelOff', () => {
        const tree = renderer
            .create(
                <UijettosToggleButton
                    labelOff="This is off"
                />
            )
            .toJSON();


        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly when given a labelOn', () => {
        const tree = renderer
            .create(
                <UijettosToggleButton
                    initialState={ true}
                    labelOn="This is on"
                />
            )
            .toJSON();


        expect(tree).toMatchSnapshot();
    });

    test('Renders correctly and button is disabled when disabled prop is set to true', () => {
        const tree = renderer
            .create(
                <UijettosToggleButton
                    disabled={ true }
                />
            )
            .toJSON();


        expect(tree).toMatchSnapshot();
    });

    describe('Stateless mode', () => {
        test('Renders correctly when state prop is set to true, ignoring the initialState', () => {
            const tree = renderer
                .create(
                    <UijettosToggleButton
                        initialState={ false }
                        state={ true }
                    />
                )
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        test('Correctly calls the whenClicked function and renders correctly after being clicked, when state prop is set to true', () => {
            const mockWhenChanged = jest.fn();
            const component = renderer
                .create(
                    <UijettosToggleButton
                        initialState={ false }
                        state={ true }
                        whenClicked={ mockWhenChanged }
                    />
                );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();

            // trigger click
            tree.props.onClick();
            expect( mockWhenChanged ).toHaveBeenCalledWith(false);
            tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });



        test('Renders correctly when state prop is set to false, ignoring the initialState', () => {
            const tree = renderer
                .create(
                    <UijettosToggleButton
                        initialState={ true }
                        state={ false }
                    />
                )
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        test('Correctly calls the whenClicked function and renders correctly after being clicked, when state prop is set to false', () => {
            const mockWhenChanged = jest.fn();
            const component = renderer
                .create(
                    <UijettosToggleButton
                        initialState={ true }
                        state={ false }
                        whenClicked={ mockWhenChanged }
                    />
                );
            let tree = component.toJSON();
            expect(tree).toMatchSnapshot();

            // trigger click
            tree.props.onClick()
            
            expect( mockWhenChanged ).toHaveBeenCalledWith(true);
            tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    })
});
