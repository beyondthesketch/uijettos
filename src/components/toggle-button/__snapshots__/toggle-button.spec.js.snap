// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UijettosToggleButton component Renders correctly and button is disabled when disabled prop is set to true 1`] = `
<button
  className="uijettos-toggle-button uijettos-toggle-button--disabled"
  disabled={true}
  onClick={[Function]}
>
  <span
    className="uijettos-toggle-button__label"
  >
    off
  </span>
</button>
`;

exports[`UijettosToggleButton component Renders correctly when given a cssClassPrefix and state is false 1`] = `
<button
  className="doodads-toggle-button"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="doodads-toggle-button__label"
  >
    off
  </span>
</button>
`;

exports[`UijettosToggleButton component Renders correctly when given a cssClassPrefix and state is true 1`] = `
<button
  className="doodads-toggle-button doodads-toggle-button--on"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="doodads-toggle-button__label doodads-toggle-button__label--on"
  >
    on
  </span>
</button>
`;

exports[`UijettosToggleButton component Renders correctly when given a labelOff 1`] = `
<button
  className="uijettos-toggle-button"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="uijettos-toggle-button__label"
  >
    This is off
  </span>
</button>
`;

exports[`UijettosToggleButton component Renders correctly when given a labelOn 1`] = `
<button
  className="uijettos-toggle-button uijettos-toggle-button--on"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="uijettos-toggle-button__label uijettos-toggle-button__label--on"
  >
    This is on
  </span>
</button>
`;

exports[`UijettosToggleButton component Renders correctly when given initialState of false 1`] = `
<button
  className="uijettos-toggle-button"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="uijettos-toggle-button__label"
  >
    off
  </span>
</button>
`;

exports[`UijettosToggleButton component Renders correctly when given initialState of true 1`] = `
<button
  className="uijettos-toggle-button uijettos-toggle-button--on"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="uijettos-toggle-button__label uijettos-toggle-button__label--on"
  >
    on
  </span>
</button>
`;

exports[`UijettosToggleButton component Renders correctly when not supplied any props 1`] = `
<button
  className="uijettos-toggle-button"
  disabled={false}
  onClick={[Function]}
>
  <span
    className="uijettos-toggle-button__label"
  >
    off
  </span>
</button>
`;
